test/Elastic.Apm.EntityFrameworkCore.Tests/ExternalDbTests.cs

```diff
@@ -0,0 +1,345 @@
+using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Linq;
+using Elastic.Apm.Api;
+using Elastic.Apm.Helpers;
+using Elastic.Apm.Model;
+using Elastic.Apm.Tests.Extensions;
+using Elastic.Apm.Tests.Mocks;
+using Elastic.Apm.Tests.TestHelpers;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using MySql.Data.MySqlClient;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Elastic.Apm.EntityFrameworkCore.Tests
+{
+	/// <summary>
+	/// Tests using external DB servers.
+	/// Tests will not run (even though they will show as passed) if any of the following environment variables is not set:
+	///  	- ELASTIC_APM_TESTS_XYZ_HOST
```

Comment:
First of all, it would be nice to have documentation for configuring the test environment because no one looks through tests to find out information configuration. It was also my mistake in https://github.com/elastic/apm-agent-dotnet/pull/616#issuecomment-557233519 and the worst thing you don't have indicates about misconfiguration.
Second, it would be nice to automate such a process, as an example, with `TestEnvironment.Docker` package, which is used by me in #592. It will allow you to forget about the configuration of your environment, you need only run docker, which is not a big deal.

Reply:
I agree that this whole area of testing agent with in real environment - real databases, real application servers, etc., should be improved - I've opened #667 to follow up.